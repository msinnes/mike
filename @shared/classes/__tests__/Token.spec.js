const Token = require('../Token');

describe('Token', () => {
  it('should be a class', () => {
    expect(() => {
      new Token('string');
    }).not.toThrow();

    expect(() => {
      Token('string');
    }).toThrowErrorMatchingSnapshot();
  });

  it('should require a type parameter', () => {
    expect(() => {
      new Token('type');
    }).not.toThrow();

    expect(() => {
      new Token();
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new Token(1);
    }).toThrowErrorMatchingSnapshot();
  });

  it('should default value to null if none is provided', () => {
    const token = new Token('type');
    expect(token.value).toEqual(null);
  });

  it('should optionally take a value argument', () => {
    const ref = {};
    const token = new Token('type', ref);
    expect(token.value).toEqual(ref);
  });
});

