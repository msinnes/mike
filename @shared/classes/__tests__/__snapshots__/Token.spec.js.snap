// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Token should be a class 1`] = `"ClassError: Cannot call a class as a function"`;

exports[`Token should require a type parameter 1`] = `"ValidationError: type must be a string"`;

exports[`Token should require a type parameter 2`] = `"ValidationError: type must be a string"`;
